nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,276 @@
+@CapabilityDescription("Extract the content metadata from flowfiles containing audio, video, image, and other file "
+        + "types.  This processor relies on the Apache Tika project for file format detection and parsing.  It "
+        + "extracts a long list of metadata types for media files including audio, video, and print media "
+        + "formats."
+        + "For the more details and the list of supported file types, visit the library's website "
+        + "at http://tika.apache.org/.")
+@WritesAttributes({@WritesAttribute(attribute = "tika.<attribute>", description = "The extracted content metadata "
+        + "will be inserted with the attribute name \"tika.<attribute>\". ")})
+@SupportsBatching
+public class ExtractMediaMetadata extends AbstractProcessor {
+
+    public static final PropertyDescriptor MAX_NUMBER_OF_ATTRIBUTES = new PropertyDescriptor.Builder()
+            .name("Max Number of Attributes")
+            .description("Specify the max number of attributes to add to the flowfile. There is no guarantee in what order"
+                    + " the tags will be processed. By default it will process all of them.")
+            .required(false)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor FILENAME_FILTER = new PropertyDescriptor.Builder()
+            .name("File Name Filter")
+            .description("A regular expression identifying file names which metadata should extracted.  As flowfiles"
+                    + " are processed, if the file name matches this regular expression or this expression is"
+                    + " blank, the flowfile will be scanned for it's MIME type and metadata.  If left blank, all"
+                    + " flowfiles will be scanned.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MIME_TYPE_FILTER = new PropertyDescriptor.Builder()
+            .name("MIME Type Filter")
+            .description("A regular expression identifying MIME types for which metadata should extracted.  Flowfiles"
+                    + " selected for scanning by the File Name Filter are parsed to determine the MIME type and extract"
+                    + " metadata.  If the MIME type found matches this regular expression or this expression is"
+                    + " blank, the metadata keys that match the Metadata Key Filter will be added to the flowfile"
+                    + " attributes.  There is no guarantee in what order attributes will be produced.  If"
+                    + " left blank, metadata will be extracted from all flow files selected for scanning.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
```

Comment:
Why can't the user filter on the mime-type already on the flowfile? If it's hit the point where Tika has has already parsed it (the most significant processing time), to get the parsed mime-type, then the only part left is to add the metadata as attributes. 

There would be a use-case where the user doesn't want to add the attributes after it's been parsed? 

The main use-case of this filter I was thinking of was to filter the mime-types that are parsed so that only the ones that will succeed are attempted. This use-case would not be achieved by this mime-type filter because it would have already attempted parsing the document.